### PR TITLE
completed level 1 and 2 of node-fs-readfile exercise

### PR DESCRIPTION
**Pull request by hsu-isaac**
image

### CODE REVIEW

File: node-fs-readfile/cat.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+console.log(process.argv);
+
+for (let i = 2; i < process.argv.length; i++) {
+  try {
+    const filename = process.argv[i];
+    fs.readFile(filename, 'utf8', (err, data) => {
+      if (err) throw err;
+      console.log(data);
+    });
+  } catch (err) {}
```
Comment (thebearingedge) [inline node-fs-readfile/cat.js:12]: Why the `try/catch`? If there's an error reading the file, unfortunately, the error will not be thrown synchronously. You will need a mind shift for this one as you cannot guarantee that the reads will complete in the order you start them.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Please include a video of the level three working. I'm leaving this review to update my records.

**Comment by hsu-isaac** [comment]
Never mind, it doesn't work - I guess I just got lucky with the tests I ran earlier

**Comment by hsu-isaac** [comment]
node-fs-readfile

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, good, you are on the right track, but we can simplify this a bit and remove a bunch of the code

- create an empty array
- start a counter
- loop through your arguments
- start a readfile for every argument
- when each one finishes
  - increment your counter
  - add the data to the array at the same index as the argument
  - check if the counter is done and if it is
    - print each value store in the array

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**